components/html-block/html-block.js

```diff
@@ -244,7 +244,7 @@ class HtmlBlock extends LitElement {
 
 	async updated(changedProperties) {
 		super.updated(changedProperties);
-		if ((changedProperties.has('html') || changedProperties.has('_context')) && this.html !== undefined && this.html !== null && !this.noDeferredRendering) {
+		if (changedProperties.has('embeds') && this.html !== undefined && this.html !== null && !this.noDeferredRendering) {
```

Comment:
It looks like `_context` doesn't ever actually change after it's set in the constructor so I don't see any updates happening from it after the initial one (where it doesn't even trigger this `if`). At least not from current test cases. And `_processRenderers` is already waiting on those.

If they can change elsewhere that I'm not seeing, perhaps we should run `this.update(new Map([['_context']]))` somewhere.

Reply:
How does Lit handle values changing in maps? https://github.com/BrightspaceUI/core/blob/main/components/html-block/html-block.js#L219

This might need more careful treatment now that `_context` is a Map instead of an object, but its values can definitely change from that callback (at any time post-construction).

Reply:
In practice this doesn't really happen. Nowhere in the LMS do those context objects ever get updated once they're first set. It's _theoretically_ possible for that to happen.

Reply:
Mutating won't trigger an update, but all this is only kicked off once from the constructor, right? If so, [this](https://github.com/BrightspaceUI/core/blob/main/components/html-block/html-block.js#L279) would seem to suggest we're ok not triggering additional updates since the `until` should ultimately handle it all.

`until` -> `_processEmbeds` -> `update` -> `_updateRenderContainer` -> `_processRenderers` -> `await _initialContextPromise`

Reply:
Ohh, I see, that `ctx => this._context.set(key, ctx)` can be called as a callback whenever, multiple times?

Reply:
That callback though (`ctx => this._context.set(key, ctx`) can be invoked arbitrarily after the fact though. It gets set in the context provider stuff and can be invoked by the client here: https://github.com/BrightspaceUI/lms-context-provider/blob/main/src/client/client-internal.js#L49

Right now it's clear that this isn't happening... And maybe that's okay and we don't need to care. Certainly right now this context object is effectively immutable in the LMS. But technically it _could_ be changed.

Reply:
Oop, you got in before me, haha. Yes, it can.